File: gin/modules/admin/sys_config.go
package admin

import (
	"webmis/config"
	"webmis/library"
	"webmis/model"
	"webmis/service"
	"webmis/util"

	"github.com/gin-gonic/gin"
)

type SysConfig struct {
	service.Base
}

/* 列表 */
func (r SysConfig) List(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, c.Request.RequestURI)
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	// 查询
	model := (&model.SysConfig{}).New()
	model.Columns("name", "val")
	data := model.Find()
	list := map[string]interface{}{}
	// 数据
	for _, val := range data {
		if val["name"] == "logo" || val["name"] == "login_bg" {
			list[val["name"].(string)] = (&service.Data{}).Img(val["val"])
		} else {
			list[val["name"].(string)] = val["val"]
		}
	}
	// 返回
	r.GetJSON(c, gin.H{"code": 0, "msg": "成功", "list": list})
}

/* 编辑 */
func (r SysConfig) Edit(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	data, _ := r.JsonName(json, "data")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, c.Request.RequestURI)
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if data == "" {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 数据
	m := (&model.SysConfig{}).New()
	param := map[string]interface{}{}
	util.JsonDecode(data, &param)
	for key, val := range param {
		if key == "logo" || key == "login_bg" {
			continue
		}
		m.Set(map[string]interface{}{"val": util.Trim(val)})
		m.Where("name=?", key)
		if !m.Update() {
			r.GetJSON(c, gin.H{"code": 5000, "msg": "更新失败!"})
			return
		}
	}
	r.GetJSON(c, gin.H{"code": 0, "msg": "成功"})
}

/* 头像 */
func (r SysConfig) Upimg(c *gin.Context) {
	// 参数
	json := map[string]interface{}{}
	c.BindJSON(&json)
	token, _ := r.JsonName(json, "token")
	name, _ := r.JsonName(json, "name")
	base64, _ := r.JsonName(json, "base64")
	// 验证
	msg := (&service.AdminToken{}).Verify(token, c.Request.RequestURI)
	if msg != "" {
		r.GetJSON(c, gin.H{"code": 4001, "msg": msg})
		return
	}
	if base64 == "" {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "参数错误!"})
		return
	}
	// 类型
	if name != "logo" && name != "login_bg" {
		r.GetJSON(c, gin.H{"code": 4000, "msg": "类型错误!"})
		return
	}
	// 上传
	ImgDir := "upload/admin/img/"
	img := (&library.Upload{}).Base64(map[string]interface{}{"path": ImgDir, "base64": base64})
	if img == "" {
		r.GetJSON(c, gin.H{"code": 5000, "msg": "上传失败!"})
		return
	}
	// 数据
	m := (&model.SysConfig{}).New()
	m.Columns("val")
	m.Where("name=?", name)
	imgData := m.FindFirst()
	m.Set(map[string]interface{}{"val": ImgDir + img})
	m.Where("name=?", name)
	if !m.Update() {
		r.GetJSON(c, gin.H{"code": 5000, "msg": "上传失败!"})
		return
	}
	// 清理
	rmImg := imgData["val"].(string)
	(&library.FileEo{}).New(config.Env().RootDir)
	(&library.FileEo{}).RemoveAll(rmImg)
	// 返回
	r.GetJSON(c, gin.H{"code": 0, "msg": "成功", "img": (&service.Data{}).Img(ImgDir + img)})
}

File: python/modules/admin/sys_config.py
from flask import request

from library.file_eo import FileEo
from library.upload import Upload
from service.base import Base
from service.data import Data
from service.admin_token import AdminToken
from model.sys_config import SysConfig as SysConfigM
from util.util import Util

class SysConfig(Base):

  ImgDir: str='upload/admin/img/'

  # 列表
  def List(self):
    # 参数
    json = self.Json()
    token = self.JsonName(json, 'token')
    # 验证
    msg = AdminToken.Verify(token, request.path)
    if msg != '' :
      return self.GetJSON({'code':4001, 'msg':msg})
    # 查询
    m = SysConfigM()
    m.Columns('name', 'val')
    data = m.Find()
    # 数据
    list = {}
    for val in data :
      if val['name']=='logo' or val['name']=='login_bg' :
        list[val['name']] = Data.Img(val['val'])
      else :
        list[val['name']] = val['val']
    # 返回
    return self.GetJSON({'code':0, 'msg':'成功', 'list':list})

  # 编辑
  def Edit(self):
    # 参数
    json = self.Json()
    token = self.JsonName(json, 'token')
    data = self.JsonName(json, 'data')
    # 验证
    msg = AdminToken.Verify(token, request.path)
    if msg != '' :
      return self.GetJSON({'code':4001, 'msg':msg})
    if not data :
      return self.GetJSON({'code':4000, 'msg':'参数错误!'})
    # 模型
    m = SysConfigM()
    param = Util.JsonDecode(data)
    for key, val in param.items() :
      if key=='logo' or key=='login_bg' : continue
      m.Set({'val': Util.Trim(val)})
      m.Where('name=%s', key)
      if not m.Update() :
        return self.GetJSON({'code':5000, 'msg':'更新失败!'})
    return self.GetJSON({'code':0, 'msg':'成功'})

  # 头像
  def Upimg(self):
    # 参数
    json = self.Json()
    token = self.JsonName(json, 'token')
    name = self.JsonName(json, 'name')
    base64 = self.JsonName(json, 'base64')
    # 验证
    msg = AdminToken.Verify(token, request.path)
    if msg != '' :
      return self.GetJSON({'code':4001, 'msg':msg})
    if not base64 :
      return self.GetJSON({'code':4000, 'msg':'参数错误!'})
    if name!='logo' and name!='login_bg' :
      return self.GetJSON({'code':4000, 'msg':'类型错误!'})
    # 上传
    img = Upload.Base64({'path':self.ImgDir, 'base64':base64})
    if not img : return self.GetJSON({'code':5000, 'msg':'上传失败!'})
    # 模型
    m = SysConfigM()
    m.Columns('val')
    m.Where('name=%s', name)
    imgData = m.FindFirst()
    m.Set({'val': self.ImgDir+img})
    m.Where('name=%s', name)
    if not m.Update() : return self.GetJSON({'code':5000, 'msg':'上传失败!'})
    # 清理
    rmImg = imgData['val']
    FileEo.RemoveAll(rmImg)
    # 返回
    return self.GetJSON({'code':0, 'msg':'成功', 'img':Data.Img(self.ImgDir+img)})